internal/impl/nats/input.go

```diff
@@ -251,6 +251,9 @@ func (n *natsReader) Read(ctx context.Context) (*service.Message, service.AckFun
 }
 
 func (n *natsReader) Close(ctx context.Context) (err error) {
+	go func() {
```

Comment:
is there any particular reason we need to wrap it inside a go routine? I think we'd better to just remove the go routine to make it a sync function.